modules/study/src/test/newTreeTest.scala

```diff
@@ -0,0 +1,167 @@
+
+  // Convertor
+  object NewTreeC:
+    def fromRoot(root: Root): Option[NewTree] =
+      root.children.first.map(first =>
+        NewTree(
+          value = NewBranchC.fromBranch(first),
+          child = first.children.first.map(NewTreeC.fromBranch),
+          variation = root.children.variations.toVariations(fromBranch)
+        )
+      )
+
+    def fromBranch(branch: Branch): NewTree =
+      NewTree(
+        value = NewBranchC.fromBranch(branch),
+        child = branch.children.first.map(NewTreeC.fromBranch),
+        variation = branch.children.variations.toVariations(fromBranch)
+      )
+  extension (newTree: NewTree)
+    def toBranch: Branch = newTree.value.toBranch(newTree.children)
+    def toBranches: Branches = Branches(
+      newTree.value.toBranch(newTree.children) :: newTree.variations.map(_.toBranch)
+    )
+  // Convertor
+  object NewRootC:
+    def fromRoot(root: Root) =
+      NewRoot(MetasC.fromNode(root), NewTreeC.fromRoot(root))
+
+  extension (newRoot: NewRoot)
+    def toRoot =
+      Root(
+        newRoot.metas.ply,
+        newRoot.metas.fen,
+        newRoot.metas.check,
+        newRoot.metas.dests,
+        newRoot.metas.drops,
+        newRoot.metas.eval,
+        newRoot.metas.shapes,
+        newRoot.metas.comments,
+        newRoot.metas.gamebook,
+        newRoot.metas.glyphs,
+        newRoot.tree.fold(Branches.empty)(_.toBranches),
+        newRoot.metas.opening,
+        newRoot.metas.clock,
+        newRoot.metas.crazyData
+      )
+
+  given Conversion[String, PgnStr] = PgnStr(_)
+  given Conversion[PgnStr, String] = _.value
+
+  test("valid tree -> newTree first move") {
+    val x       = PgnImport("1. e4 *", Nil).toOption.get
+    val newRoot = NewRootC.fromRoot(x.root)
+    assertEquals(newRoot.tree.get.size, 1L)
+    assertEquals(newRoot.tree.get.mainline.map(_.move.san.value), List("e4"))
+    assertEquals(newRoot.toRoot, x.root)
+  }
+
+  test("valid tree -> newTree first move with variation") {
+    val x       = PgnImport("1. e4 (1. d4??) *", Nil).toOption.get
+    val newRoot = NewRootC.fromRoot(x.root)
+    assertEquals(newRoot.tree.get.size, 2L)
+    assertEquals(newRoot.tree.get.variations.map(_.value.move.san.value), List("d4"))
+    assertEquals(newRoot.toRoot, x.root)
+  }
+
+  test("valid tree -> newTree two moves") {
+    val x       = PgnImport("1. e4 e6 *", Nil).toOption.get
+    val newRoot = NewRootC.fromRoot(x.root)
+    assertEquals(newRoot.tree.get.size, 2L)
+    assertEquals(newRoot.tree.get.mainline.map(_.move.san.value), List("e4", "e6"))
+    assertEquals(newRoot.toRoot, x.root)
+  }
+
+  test("valid tree <-> newTree more realistic conversion"):
+    PgnFixtures.all foreach { pgn =>
+      val x = PgnImport(pgn, Nil).toOption.get
+      val y = NewPgnImport(pgn, Nil).toOption.get
+      assertEquals(y.end, x.end)
+      assertEquals(y.variant, x.variant)
+      assertEquals(y.tags, x.tags)
+      assertEquals(y.root.toRoot, x.root)
```

Comment:
This tests only works if generated Comment.Id are the same, we will need to think about how to test this.